src/components/PostsList.tsx

```diff
@@ -1,85 +1,66 @@
 import React from 'react';
+import { Post } from '../types/Post';
 
-export const PostsList: React.FC = () => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+type Props = {
+  posts: Post[] | null;
+  targetPost: Post | null;
+  isOpen: boolean;
+  setTargetPost: (post: Post) => void;
+  setIsOpen: (val: boolean) => void;
+};
 
-    <table className="table is-fullwidth is-striped is-hoverable is-narrow">
-      <thead>
-        <tr className="has-background-link-light">
-          <th>#</th>
-          <th>Title</th>
-          <th> </th>
-        </tr>
-      </thead>
+export const PostsList: React.FC<Props> = ({
+  posts,
+  targetPost,
+  isOpen,
+  setTargetPost,
+  setIsOpen,
+}) => {
+  const hendleClick = (post: Post) => {
+    setTargetPost(post);
+    setIsOpen(!isOpen);
+  };
 
-      <tbody>
-        <tr data-cy="Post">
-          <td data-cy="PostId">17</td>
+  return (
+    <div data-cy="PostsList">
+      <p className="title">Posts:</p>
 
-          <td data-cy="PostTitle">
-            fugit voluptas sed molestias voluptatem provident
-          </td>
+      <table className="table is-fullwidth is-striped is-hoverable is-narrow">
+        <thead>
+          <tr className="has-background-link-light">
+            <th>#</th>
+            <th>Title</th>
+            <th> </th>
+          </tr>
+        </thead>
 
-          <td className="has-text-right is-vcentered">
-            <button
-              type="button"
-              data-cy="PostButton"
-              className="button is-link is-light"
-            >
-              Open
-            </button>
-          </td>
-        </tr>
+        <tbody>
+          {posts && posts.map(post => (
```

Comment:
```suggestion
          {posts && posts.map(post => (
```
Move this check to higher component because there is no sense in render PostsList component if we don't have a posts